app/models/news_item.rb

```diff
@@ -1,15 +1,7 @@
 class NewsItem < ActiveRecord::Base
   extend Enumerize
   enumerize :state, in: [:draft, :archived, :published]
-  validates :published_at, presence: true, if: :published?
+  scope :published, -> { where(state: :published).order(:published_at) }
+  validates :published_at, presence: true, if: lambda{ state.published? } 
```

Comment:
@itsWill Thanks for making this change. One suggestion would be to use consistent lambda syntax. In the scope, the "stabby lambda" `->` (preferred), and in the validation the keyword `lambda` is used.